### DISCUSSION TITLE
Issues with Timezone Configuration for Valhalla

### DISCUSSION THREAD

**Discussion by pipo3944** [valhalla/valhalla#4763] [Q&A] [answered]
Hi Valhalla community,

I have been testing Valhalla with GTFS data and have encountered issues with the time settings.

Specifically, I have loaded GTFS data that only operates from around 07:00 to 20:00. However, when performing route searches in the area covered by this GTFS data, I encountered the following issues:

**1. When specifying 12:00:**
```
    "date_time": {
      "type": 1,
      "value": "2024-06-13T12:00"
    }
 ```
No results were returned.
```
{"error_code":442,"error":"No path could be found for input","status_code":400,"status":"Bad Request"}
```

**2. When specifying 00:00:**
```
    "date_time": {
      "type": 1,
      "value": "2024-06-13T00:00"
    }
```
The response shows bus travel at 18:00.
<details><summary>Result</summary>
<pre>
{
  "trip": {
    "locations": [
      {
        "type": "break",
        "lat": 34.865479,
        "lon": 138.310942,
        "date_time": "2024-06-13T08:35",
        "original_index": 0
      },
      {
        "type": "break",
        "lat": 34.849124,
        "lon": 138.284139,
        "side_of_street": "left",
        "original_index": 1
      }
    ],
    "legs": [
      {
        "maneuvers": [
          {
            "type": 1,
            "instruction": "Walk west.",
            "verbal_succinct_transition_instruction": "Walk west.",
            "verbal_pre_transition_instruction": "Walk west.",
            "verbal_post_transition_instruction": "Continue for 10 meters.",
            "time": 8.401,
            "length": 0.011,
            "cost": 8.401,
            "begin_shape_index": 0,
            "end_shape_index": 1,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 30,
            "instruction": "Take the 焼津循環線　さつき toward 焼津駅前. (7 stops)",
            "verbal_pre_transition_instruction": "Take the 焼津循環線　さつき toward 焼津駅前.",
            "verbal_post_transition_instruction": "Travel 7 stops.",
            "time": 787.0,
            "length": 3.039,
            "cost": 5491.92,
            "begin_shape_index": 1,
            "end_shape_index": 16,
            "depart_instruction": "Depart: 18:03 from 西小学校前.",
            "verbal_depart_instruction": "Depart at 18:03 from 西小学校前.",
            "arrive_instruction": "Arrive: 18:14 at 柳新屋.",
            "verbal_arrive_instruction": "Arrive at 18:14 at 柳新屋.",
            "transit_info": {
              "onestop_id": "Yaizushi-GTFS-JP_102003",
              "long_name": "焼津循環線　さつき",
              "headsign": "焼津駅前",
              "color": 16711680,
              "text_color": 16777215,
              "operator_onestop_id": "Yaizushi-GTFS-JP_1000020222127",
              "operator_name": "焼津市",
              "operator_url": "http://www.city.yaizu.lg.jp/g06-002/kurashi/koutsuu/bus/index.html",
              "transit_stops": [
                {
                  "type": "stop",
                  "onestop_id": "Yaizushi-GTFS-JP_135_02_transit_station",
                  "name": "西小学校前",
                  "departure_date_time": "2024-06-13T18:03+09:00",
                  "assumed_schedule": false,
                  "lat": 34.865599,
                  "lon": 138.311015
                },
                ...
              ]
            },
            "travel_mode": "transit",
            "travel_type": "bus"
          },
          ...
</pre>
</details> 

**3. When specifying 05:00:**
```
"date_time": {
  "type": 1,
  "value": "2024-06-13T05:00"
}
```
The response indicates bus travel at 23:00.
<details><summary>Result</summary>
<pre>
...
          {
            "type": 24,
            "instruction": "Keep left at the fork.",
            "verbal_transition_alert_instruction": "Keep left at the fork.",
            "verbal_pre_transition_instruction": "Keep left at the fork.",
            "verbal_post_transition_instruction": "Continue for 20 meters.",
            "time": 12.0,
            "length": 0.017,
            "cost": 18.0,
            "begin_shape_index": 48,
            "end_shape_index": 50,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 30,
            "instruction": "Take the 焼津循環線　ゆりかもめ toward 焼津駅前. (1 stop)",
            "verbal_pre_transition_instruction": "Take the 焼津循環線　ゆりかもめ toward 焼津駅前.",
            "verbal_post_transition_instruction": "Travel 1 stop.",
            "time": 1594.0,
            "length": 0.685,
            "cost": 2818.921,
            "begin_shape_index": 50,
            "end_shape_index": 60,
            "depart_instruction": "Depart: 23:45 from 泓の川.",
            "verbal_depart_instruction": "Depart at 23:45 from 泓の川.",
            "arrive_instruction": "Arrive: 23:47 at 三右衛門新田.",
            "verbal_arrive_instruction": "Arrive at 23:47 at 三右衛門新田.",
            "transit_info": {
              "onestop_id": "Yaizushi-GTFS-JP_101003",
              "long_name": "焼津循環線　ゆりかもめ",
              "headsign": "焼津駅前",
              "color": 28864,
              "text_color": 16777215,
              "operator_onestop_id": "Yaizushi-GTFS-JP_1000020222127",
              "operator_name": "焼津市",
              "operator_url": "http://www.city.yaizu.lg.jp/g06-002/kurashi/koutsuu/bus/index.html",
              "transit_stops": [
                {
                  "type": "stop",
                  "onestop_id": "Yaizushi-GTFS-JP_112_01_transit_station",
                  "name": "泓の川",
                  "departure_date_time": "2024-06-13T23:45+09:00",
                  "assumed_schedule": false,
                  "lat": 34.854154,
                  "lon": 138.301316
                },
                {
                  "type": "stop",
                  "onestop_id": "Yaizushi-GTFS-JP_113_01_transit_station",
                  "name": "三右衛門新田",
                  "arrival_date_time": "2024-06-13T23:47+09:00",
                  "assumed_schedule": false,
                  "lat": 34.848105,
                  "lon": 138.29989
                }
              ]
            },
            "travel_mode": "transit",
            "travel_type": "bus"
          },
...
</pre>
</details> 

---
I have attempted to configure the timezone for Valhalla using the following steps:

1. Set locale:
```
sudo locale-gen ja_JP.UTF-8
export LANG=ja_JP.UTF-8
export LANGUAGE=ja_JP:ja
export LC_ALL=ja_JP.UTF-8
```
2. Build timezone data:
```
valhalla_build_timezones > /custom_files/timezone_data/timezones.sqlite
```

Am I missing any steps, or is there an error in my approach?
Any advice or insights would be greatly appreciated.

Thank you!

**Comment by kevinkreiser** [comment, marked as answer]
the timezone data needs to be avaialble (and configured in the json config file) at the time you built the routing tiles, was that the case? if so then all i can say is that transit has a number of issues we are still working out. sadly no one has a lot of time to do it these days so there is very little progress lately

**Comment by pipo3944** [comment]
@kevinkreiser 
Ah, I apologize for that! As you mentioned, I reconfigured the time zone settings and rebuilt everything from scratch. It seems to be working correctly now.

Thank you for your assistance!
